spec/models/ticket_spec.rb

```diff
@@ -330,4 +330,141 @@
       end
     end
   end
+
+  describe '#access_condition' do
+
+    it 'checks user permission' do
+      user = create(:agent_user)
+      expect(Ticket.access_condition(user, 'overview')[0]).to eq 'group_id IN (?)'
+      user = create(:customer_user)
+      expect(Ticket.access_condition(user, 'overview')[0]).not_to eq 'group_id IN (?)'
+    end
+
+    it 'checks user has Primary Organization' do
+      user = create(:customer_user)
+      expect(Ticket.access_condition(user, 'overview')[0]).not_to eq '(tickets.customer_id = ? OR tickets.organization_id = ?)'
+      organization = create(:organization)
+      user = create(:customer_user, organization: organization)
+      expect(Ticket.access_condition(user, 'overview')[0]).to eq '(tickets.customer_id = ? OR tickets.organization_id = ?)'
+    end
+
+    it 'checks user has Alternative Organizations' do
+      user = create(:customer_user)
+      expect(Ticket.access_condition(user, 'overview')[0]).not_to eq '(tickets.customer_id = ? OR tickets.organization_id IN (?))'
+      organization = create(:organization)
+      user = create(:customer_user, organizations: [organization])
+      expect(Ticket.access_condition(user, 'overview')[0]).to eq '(tickets.customer_id = ? OR tickets.organization_id IN (?))'
+    end
+
+    it 'returns if no organization or not agent' do
+      user = create(:customer_user)
+      expect(Ticket.access_condition(user, 'overview')[0]).to eq 'tickets.customer_id = ?'
+    end
+  end
+
+  describe '#check_defaults' do
```

Comment:
Please don't test private/internal methods because they should be easy to change. Instead test functionality. You might want to use [run_callbacks](http://devdocs.io/rails~5.1/activesupport/callbacks#method-i-run_callbacks) instead.